lib/screens/authentication/recover/recover_account_timer/interactor/viewmodels/recover_account_timer_bloc.dart

```diff
@@ -32,16 +34,20 @@ class RecoverAccountTimerBloc extends Bloc<RecoverAccountTimerEvent, RecoverAcco
 
   Future<void> _fetchInitialData(FetchTimerData event, Emitter<RecoverAccountTimerState> emit) async {
     emit(state.copyWith(pageState: PageState.loading));
-    final Result<int> result = await FetchRecoverAccountTimerData().run(state.recoveryModel);
+    final Result<DateTime> result =
+        await FetchRecoverAccountTimerDataUseCase().run(state.recoveryModel, state.configModel);
 
     if (result.isValue) {
-      final data = result.asValue!.value;
-      emit(state.copyWith(pageState: PageState.success, timeLockExpirySeconds: data));
+      final expirationDate = result.asValue!.value;
+      print("expiration date: $expirationDate");
 
-      if (data != 0) {
+      emit(state.copyWith(pageState: PageState.success, timeLockExpirationDate: expirationDate));
+
+      if (state.timeRemainingSeconds > 0) {
         await _tickerSubscription?.cancel();
         _tickerSubscription = _tick().listen((timer) => add(Tick(timer)));
       }
```

Comment:
added an emit here to update the timer before it is shown to the user

Otherwise it would show 0:00:00 for 1 second until the ticker kicks in.